### DISCUSSION TITLE
Is there a "Python Version" of retrieval functions and automation functions

### DISCUSSION THREAD

**Discussion by pydaxing** [microsoft/semantic-kernel#7095] [General]
Is there a "Python Version" of "Using data retrieval functions for RAG" and "Using task automation functions" in https://learn.microsoft.com/en-us/semantic-kernel/concepts/plugins/using-task-automation-functions

**Comment by nmoeller** [comment]
@pydaxing i am not sure what the Status is in the Learning, but if you are just looking for samples check out the concepts in the repo.

[Python RAG Example](https://github.com/microsoft/semantic-kernel/tree/main/python/samples/concepts/rag)
[Python Task Automation Function](https://github.com/microsoft/semantic-kernel/blob/main/python/samples/concepts/filtering/function_invocation_filters.py)

Hope that helps, if there is anything feel free to ask !
